zerver/forms.py

```diff
@@ -210,8 +211,11 @@ def send_mail(self, subject_template_name, email_template_name,
         the mail through the functions in zerver.lib.send_email, to match
         how we send all other mail in the codebase.
         """
+        request = getattr(self, 'request')
         user = get_user_profile_by_email(to_email)
-        attempted_subdomain = get_subdomain(getattr(self, 'request'))
+        attempted_subdomain = get_subdomain(request)
+        if settings.EMAIL_RATE_LIMITING:
+            rate_limit_email(request, to_email, attempted_subdomain)
```

Comment:
Also this function shouldn't get a rate-limit call -- like @timabbott said, the idea here is to prevent an attacker from attempting a lot of passwords. We'll have a rate limit per email the potential-attacker is trying to log in as, but it's not related to sending email.